source/tests/system/nidcpower_driver_api_tests.cpp

```diff
@@ -300,8 +314,15 @@ class NiDCPowerDriverApiTest : public ::testing::Test {
   DeviceServerInterface* device_server_;
   std::unique_ptr<::nidevice_grpc::Session> driver_session_;
   std::unique_ptr<dcpower::NiDCPower::Stub> nidcpower_stub_;
+  bool is_legacy_session_{false};
 };
 
+// Some tests don't work with the legacy InitializeWithChannels API
+// because of differences in how channels are configured OR because they
+// test features of the independent channel API.
+#define GTEST_SKIP_IF_LEGACY() \
```

Comment:
GTEST_SKIP includes an early return. So it needs to be a macro so we return from the test method (otherwise the later failures override the skip result).